zerver/lib/presence.py

```diff
@@ -147,14 +147,24 @@ def get_presence_for_user(
 
 
 def get_presence_dict_by_realm(
-    realm: Realm, slim_presence: bool = False, requesting_user_profile: Optional[UserProfile] = None
-) -> Dict[str, Dict[str, Any]]:
+    realm: Realm,
+    slim_presence: bool = False,
+    last_update_id: Optional[int] = None,
+    requesting_user_profile: Optional[UserProfile] = None,
+) -> Tuple[Dict[str, Dict[str, Any]], Optional[int]]:
     two_weeks_ago = timezone_now() - timedelta(weeks=2)
+    kwargs = dict()
+    if last_update_id is not None:
+        kwargs["last_update_id__gt"] = last_update_id
+
     query = UserPresence.objects.filter(
         realm_id=realm.id,
+        # TODO: Should we remove this check when last_update_id
+        # is present, even if =-1?
         last_connected_time__gte=two_weeks_ago,
```

Comment:
Yeah, I think we can do
```
    if last_update_id is not None:
        kwargs["last_update_id__gt"] = last_update_id
    else:
        kwargs["last_connected_time__gte"] = two_weeks_ago
```

Reply:
@timabbott Hmm I still wonder about this one. We nay want clients who use this api to send `last_update_id=-1` at startup to get all data, rather than not passing the param to do `last_update_id=None`; to allow the server to know that it's dealing a `last_update_id`-utilizing client. Even if right now we're not actually changed any behavior based on it.

So the question this brings is - should such a modern client initializing with `last_update_id=-1` get ALL presence data or just the data with the `last_connected_time__gte=two_weeks_ago` filter?

Of course if we want the two weeks limit on client initialization, we could just do 
```
    if last_update_id not in [None, -1]:
        kwargs["last_update_id__gt"] = last_update_id
    else:
        kwargs["last_connected_time__gte"] = two_weeks_ago
```

Reply:
I think what we have is pretty reasonable -- seems like `last_update_id=-1` should be a good way to ask for "no really, give me all the data". But worth flagging this detail for API design discussion.

Reply:
https://chat.zulip.org/#narrow/stream/378-api-design/topic/presence.20rewrite.20.3E.20how.20much.20data.20to.20fetch.20at.20initialization